theano/gpuarray/nerv.py

```diff
@@ -147,17 +147,18 @@ def c_cleanup_code_struct(self, node, nodename):
         return '\n'.join(codel)
 
 
-@opt.register_opt()
+@opt.register_opt('fast_compile')
 @opt.op_lifter([tensor.Dot])
-def local_dot_to_gemm16(node, ctx_name):
+@opt.register_opt2([tensor.Dot], 'fast_compile')
+def local_gpua_dot_to_gemm16(op, ctx_name, inputs, outputs):
     if nerv is None:
         return
-    A = node.inputs[0]
-    B = node.inputs[1]
+    A = inputs[0]
+    B = inputs[1]
     if (A.ndim == 2 and B.ndim == 2 and
             A.dtype == 'float16' and B.dtype == 'float16'):
-        fgraph = node.inputs[0].fgraph
-        C = GpuAllocEmpty(dtype='float16', context_name=ctx_name)(
+        fgraph = getattr(outputs[0], 'fgraph', None)
```

Comment:
we can merge as is, but why the getattr? I think with outputs, you always have the fgraph.